velox/common/file/File.cpp

```diff
@@ -377,6 +377,21 @@ void LocalWriteFile::write(
 void LocalWriteFile::truncate(int64_t newSize) {
   checkNotClosed(closed_);
   VELOX_CHECK_GE(newSize, 0, "New size cannot be negative.");
+#ifdef linux
+  if (newSize > size_) {
```

Comment:
I think we shall do this after truncate? thanks!

Reply:
I'm not sure if we need a separate truncate call when using fallback:
From fallocate man page:
> The default operation (i.e., mode is zero) of fallocate()
       allocates the disk space within the range specified by offset and
       len.  The file size (as reported by [stat(2)](https://man7.org/linux/man-pages/man2/stat.2.html)) will be changed if
       offset+len is greater than the file size.